### PR TITLE
new resource `azurerm_graph_account`

### PR DESCRIPTION
**Pull request by teowa**
reference:
- [doc](https://learn.microsoft.com/en-us/graph/metered-api-setup?tabs=azurecloudshell#enable-an-application)
- [swagger](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/graphservicesprod/resource-manager/Microsoft.GraphServices/stable/2023-04-13/graphservicesprod.json)

### CODE REVIEW

File: internal/services/graph/graph_account_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-graph-%d"
+  location = %q
+}
+
+resource "azuread_application" "test" {
```
Comment (teowa) [inline internal/services/graph/graph_account_resource_test.go:157]: this resource is depend on an `azuread_application`, however through my local test the resource creation may fail occationally with error: `performing AccountsCreateAndUpdate: unexpected status 400 with error: BadRequest: The application was not found in this tenant.`. The service team told that this is hard to change and advices to use a predefined azure ad app, can we do this?

File: internal/services/graph/graph_account_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r AccountTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_graph_account" "test" {
+  name                = "acctesta-%[2]d"
+  app_id              = azuread_application.test.application_id
```
Comment (teowa) [inline internal/services/graph/graph_account_resource_test.go:112]: this resource is depend on an `azuread_application`, however through my local test the resource creation may fail occationally with error even after the ad app is created: `performing AccountsCreateAndUpdate: unexpected status 400 with error: BadRequest: The application was not found in this tenant.`. The service team told that this is hard to change and advices to use a predefined azure ad app, can we do this?

**Comment by teowa** [comment]
Hi @manicminer , thanks for reviewing this. I have modified the testcases, but we cannot create two graph account for  single ad application, so I have made the test running in serial.
```
TF_ACC=1 go test -v ./internal/services/graph -parallel 50 -test.run=TestAccGraph -timeout 1440m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccGraphAccount
=== RUN   TestAccGraphAccount/account
=== RUN   TestAccGraphAccount/account/requiresImport
=== RUN   TestAccGraphAccount/account/basic
=== RUN   TestAccGraphAccount/account/update
=== RUN   TestAccGraphAccount/account/complete
--- PASS: TestAccGraphAccount (586.13s)
    --- PASS: TestAccGraphAccount/account (586.13s)
        --- PASS: TestAccGraphAccount/account/requiresImport (150.75s)
        --- PASS: TestAccGraphAccount/account/basic (144.59s)
        --- PASS: TestAccGraphAccount/account/update (150.11s)
        --- PASS: TestAccGraphAccount/account/complete (140.68s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/graph 587.394s

```

**Comment by teowa** [comment]
Hi @manicminer, kindly ping. The service team is asking for this.

**Review by manicminer — CHANGES_REQUESTED** [review]
Thanks @teowa, this mostly LGTM. If you can look at the comments below then this should be good to merge.

Tests look good for me:

Screenshot 2023-07-13 at 13 04 23

**Pull request merged**